adapters/Vungle/VungleAdapter/VungleRouter.m

```diff
@@ -127,22 +211,69 @@ - (BOOL)playAd:(UIViewController *)viewController
                              options:options
                          placementID:delegate.desiredPlacement
                                error:&error]) {
-    startPlaying = false;
+    didAdStartPlaying = false;
   }
   if (error) {
     NSLog(@"Adapter failed to present ad, error %@", [error localizedDescription]);
-    startPlaying = false;
-  };
-  return startPlaying;
+    didAdStartPlaying = false;
+  }
+
+  return didAdStartPlaying;
+}
+
+- (UIView *)renderBannerAdInView:(UIView *)bannerView
+                        delegate:(id<VungleDelegate>)delegate
+                          extras:(VungleAdNetworkExtras *)extras
+                  forPlacementID:(NSString *)placementID {
+  NSError *bannerError = nil;
+  NSMutableDictionary *options = [[NSMutableDictionary alloc] init];  ///
+  if (extras.muted) {
+    [VungleSDK sharedSDK].muted = extras.muted;
+  } else {
+    [VungleSDK sharedSDK].muted = YES;
+  }
+  if (extras.userId) [options setObject:extras.userId forKey:VunglePlayAdOptionKeyUser];
+  if (extras.ordinal) [options setObject:@(extras.ordinal) forKey:VunglePlayAdOptionKeyOrdinal];
+  if (extras.flexViewAutoDismissSeconds)
+    [options setObject:@(extras.flexViewAutoDismissSeconds)
+                forKey:VunglePlayAdOptionKeyFlexViewAutoDismissSeconds];
+
+  BOOL success = [[VungleSDK sharedSDK] addAdViewToView:bannerView
+                                            withOptions:options
+                                            placementID:placementID
+                                                  error:&bannerError];
```

Comment:
@joshuajplagonera As you pointed out, there could be a case in which SDK returns `NO` but `error` is still `nil`. This will require SDK update. We will address this issue in next version of our SDK. Hope this won't block PR approval.

Reply:
Thanks for letting us know @ashinagawa . 
However, we are still having issues with rendering ads on our end since `addAdViewToView` is returning `NO` on my end, so we're not able to properly verify functionality yet.

Reply:
We're able to receive fill from our end now. Resolving...